File: testdata/fixtures/python/vulnerable/header_injection.py
from django.http import HttpResponse

def set_header(request):
    response = HttpResponse("OK")
    # VULNERABLE: setting header from request input without CRLF sanitization
    response['X-Custom'] = request.GET['custom']
    response['X-User-Token'] = request.POST['token']
    return response

File: testdata/fixtures/java/vulnerable/HeaderInjection.java
import javax.servlet.http.*;

public class HeaderInjection extends HttpServlet {
    protected void doGet(HttpServletRequest request, HttpServletResponse response) {
        // VULNERABLE: setting header from request parameter without CRLF sanitization
        String customVal = request.getParameter("val");
        response.setHeader("X-Custom", request.getParameter("val"));
        response.addHeader("X-User", request.getParameter("user"));
    }
}
